### ISSUE TITLE
Amazon Fire TV 4k beta1 release not prompted for port number

### ISSUE THREAD

**Issue by NeverEnoughTime** [jellyfin/jellyfin-androidtv#36]
Just installed the amazon beta1 release on grandma's Fire TV 4K stick hoping she could connect to our jellyfin server but it doesn't prompt for port number anywhere.  I tried adding the custom port number at the end of our address as well (media.domain.com:80) and it didn't connect that way either.

**Comment by thornbill** [comment]
Hi @NeverEnoughTime, I have noticed some unusual behavior with entering the url also. We should definitely work on improving this experience. As a workaround, you should be able to enter the url as `http://media.domain.com` and it _should_ connect.

**Comment by NeverEnoughTime** [comment]
Thanks for the suggestion @thornbill, I tried it with http://<etc> and https://<etc> and neither worked.

**Comment by thornbill** [comment]
How do you typically connect to Jellyfin? Is it available on port 80? Do you use SSL? Just trying to narrow this down a bit.

There is a check in the code that if the url does not contain a `:` then uses port 8096 which explains why `http://media.domain.com` works for me but just `media.domain.com` does not.

**Comment by NeverEnoughTime** [comment]
Internally, the default port, but it is behind a reverse proxy from nginx with lets encrypt. It works in browsers with http://media.domain.com (80) and https://media.domain.com (443).  We use DLNA on the Bluray player, the app on our phones (browser before the app of course), and Firefox on my laptop.

**Comment by ddurdle** [comment]
pretty critical to fix up before pushing to the app store, otherwise may get lots of negative first-use experience.

**Comment by anthonylavado** [comment]
Hey all, I think this might be working now?

Could you try entering the URL in this form?

`http://media.example.com:5555`
(replace the domain and port as needed)

**Comment by NeverEnoughTime** [comment]
I won't be by Grandma's house for a couple more days, I had just stopped in this morning, but I will update her version and check it out before Monday for sure.

**Comment by NeverEnoughTime** [comment]
Finally made it over to grandmas house, as noted before, Jellyfin is on the default port number (not forwarded by the router) but behind nginx reverse proxy, port 80 redirects to 443 with SSL.  I confirmed I can connect in web browsers by using http://media.mydomain.us and https://media.mydomain.us.  On the Fire TV 4k, with the app from the store, I tried http://media.mydomain.us and it was a no go, I added :80 which also failed, however when I tried https://media.mydomain.us it connected successfully without adding any additional port options.

**Comment by rlsmntr** [comment]
Not sure if this is related but I am receiving errors on my android tv.  I always use a non standard port on my reverse proxy.  So my web browser and android phone work fine. Trying to use the android tv it fails for me. I have tried https://jellyfin.chickenhut.com:18234 and it does not work. This is the same all my other apps use. It just pops up saying "server unavailable". I have what i believe is a log showing what happens. Some things to note. I am connecting via https this shows http. It also does not list the port so I believe this is why its failing. I have removed device id, domain and apikey.

https://pastebin.com/raw/TWwk7E3i
